Xamarin.Forms.Platform.Android/AppCompat/NavigationPageRenderer.cs

```diff
@@ -237,7 +246,7 @@ protected override void OnElementChanged(ElementChangedEventArgs<NavigationPage>
 				navController.RemovePageRequested += OnRemovePageRequested;
 
 				// If there is already stuff on the stack we need to push it
-				((INavigationPageController)e.NewElement).StackCopy.Reverse().ForEach(p => PushViewAsync(p, false));
+				navController.StackCopy.Reverse().ForEach(p => PushViewAsync(p, false));
```

Comment:
I know you didn't add this code, but we might consider replacing the `.ForEach` with a `foreach` to see if it is causing a memory leak.